### PR TITLE
Fixing DHT InputPullUp to Input only

### PR DESCRIPTION
**Pull request by Ellerbach**
Changing InputPullUp to Input only
Rational:
- The InputPullUp is not supported on all platforms
- The InputPullUp is not required if the hardware includes a pull up
  - Most hardware come with the pull up already build in
  - Our recommendation from the sample is to use a physical resistor pull up. See https://github.com/dotnet/iot/tree/master/src/devices/Dhtxx/samples
  - In general, it's better to have real physical hardware pull up as you're sure it's always there

Should fix as well #1270

### CODE REVIEW

**Comment by pgrawehr** [comment]
For #1270 this is more a workaround, is it?

**Review by pgrawehr — APPROVED** [review]

**Comment by Ellerbach** [comment]
> For #1270 this is more a workaround, is it?

Yes and no. It's really not needed to use Pull Up. In general, our bindings should stick to Output and Input only. Nothing else. This is the only way to make sure it works everywhere. Hardware pull up is the way to go for any scenario.

**Review by krwq — APPROVED** [review]

File: src/devices/Dhtxx/DhtBase.cs [hunk 1]
```diff
@@ -162,7 +162,7 @@ internal virtual void ReadThroughOneWire()
             // wait 20 - 40 microseconds
             DelayHelper.DelayMicroseconds(30, true);
 
-            _controller.SetPinMode(_pin, PinMode.InputPullUp);
+            _controller.SetPinMode(_pin, PinMode.Input);
```
Comment (krwq) [inline src/devices/Dhtxx/DhtBase.cs:165]: the other way to ensure this works everywhere is have a field somewhere in this class:
PinMode _pinMode; with value `_pinMode = _controller.IsPinModeSupported(PinMode.InputPullUp) ? PinMode.InputPullUp : PinMode.Input;`

Comment (Ellerbach) [reply src/devices/Dhtxx/DhtBase.cs:165]: Yes, that's actually a good idea. Made the changes.

**Review by krwq — APPROVED** [review]

**Pull request merged**